pkg/parser/parser.go

```diff
@@ -441,8 +481,53 @@ func populateUniqueLabels(task *api.Task) error {
 		labels = append(labels, "pipe")
 	}
 
+	// Environment-specific labels
+	for key, value := range task.Environment {
+		labels = append(labels, fmt.Sprintf("%s:%s", key, value))
+	}
+
+	// Sort labels to make them comparable
+	sort.Strings(labels)
+
 	// Update task
 	task.Metadata.UniqueLabels = labels
 
 	return nil
 }
+
+func removeCommonLabels(tasks []*api.Task) {
```

Comment:
Can `populateUniqueLabels` create the correct set of labels in the first place rather then here trying to clean it up?

Reply:
I thought about that too, but the problem these functions are trying to solve is two-pass its nature because you can't know which labels will be assigned in advance, so you assign them first and then optimize.

Do you think moving `populateUniqueLabels()` functionality into `removeCommonLabels()` and renaming it to reflect that afterwards would be a better idea?

Reply:
I think `populateUniqueLabels` should accept all tasks and not just one and populate unique labels for all of them similar to how `TaskLabelUtils#uniqueLabels` works in the cloud.

Reply:
I see, does https://github.com/cirruslabs/cirrus-cli/pull/113/commits/fc005e3385e8c469f8effc8b5e954ff60aac88ea work for you?

Reply:
Yes 😍